src/diffusers/models/transformers/transformer_sd3.py

```diff
@@ -317,6 +320,13 @@ def forward(
         encoder_hidden_states = self.context_embedder(encoder_hidden_states)
 
         for index_block, block in enumerate(self.transformer_blocks):
+            # Skip specified layers
+            if skip_layers is not None and index_block in skip_layers:
+                if block_controlnet_hidden_states is not None and block.context_pre_only is False:
```

Comment:
can we skip the block of code that needs to be skipped instead of adding duplicated code here
otherwise, if we have to change this part of the code that handles controlnet residual in the future, we have to remember to change both places, which is not great